File: backend/app/schemas/meeting.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from app.schemas.participant import AllowedParticipantCreate, AllowedParticipantResponse

class MeetingCreate(BaseModel):
    title: str = Field(..., min_length=1, max_length=255)
    description: Optional[str] = None
    scheduled_start: Optional[datetime] = None
    duration_minutes: int = Field(default=30, ge=5, le=480)
    is_restricted: bool = False
    password: Optional[str] = None
    allowed_participants: Optional[List[AllowedParticipantCreate]] = None

class MeetingResponse(BaseModel):
    id: int
    meeting_code: str
    title: str
    description: Optional[str] = None
    host_id: int
    scheduled_start: datetime
    duration_minutes: int
    status: str
    is_restricted: bool
    invite_link: str
    created_at: datetime
    host_name: Optional[str] = None
    has_password: bool = False
    allowed_participants_count: int = 0
    allowed_participants: Optional[List[AllowedParticipantResponse]] = None

    class Config:
        from_attributes = True

class MeetingStatusUpdate(BaseModel):
    status: str = Field(..., pattern="^(scheduled|active|ended)$")

File: frontend/src/types/meeting.ts
export interface User {
  id: number;
  name: string;
  email: string;
  avatar_url?: string | null;
  created_at?: string;
}

export interface AllowedParticipant {
  id: number;
  meeting_id: number;
  name: string;
  email: string;
  phone?: string | null;
  has_joined: boolean;
  joined_at?: string | null;
}

export interface Meeting {
  id: number;
  meeting_code: string;
  title: string;
  description?: string | null;
  host_id: number;
  scheduled_start: string;
  duration_minutes: number;
  status: "scheduled" | "active" | "ended";
  is_restricted: boolean;
  invite_link: string;
  created_at: string;
  host_name?: string;
  has_password?: boolean;
  allowed_participants_count?: number;
  allowed_participants?: AllowedParticipant[];
}

export interface VerifyAccessResponse {
  allowed: boolean;
  join_type?: "host" | "verified" | "guest" | null;
  requires_password: boolean;
  display_name: string;
  error?: string | null;
  meeting_title?: string | null;
}

export interface ChatMessage {
  id: string;
  senderId: string;
  senderName: string;
  text: string;
  timestamp: string;
  isSelf?: boolean;
}

export interface RoomParticipant {
  clientId: string;
  displayName: string;
  isHost: boolean;
  isAudioMuted: boolean;
  isVideoMuted: boolean;
  isScreenSharing?: boolean;
  stream?: MediaStream | null;
}

export interface UploadResult {
  participants: { name: string; email: string; phone?: string | null }[];
  total_parsed: number;
  duplicates_skipped: number;
  filename: string;
}
